torch/testing/_internal/common_utils.py

```diff
@@ -2502,8 +2485,15 @@ def make_fullrank_matrices_with_distinct_singular_values(*shape, device, dtype,
         # TODO: improve the handling of complex tensors here
         real_dtype = t.real.dtype if t.dtype.is_complex else t.dtype
         k = min(shape[-1], shape[-2])
-        s = torch.arange(1., k + 1, dtype=real_dtype, device=device).mul_(1.0 / (k + 1))
-        x = (u * s.to(dtype)) @ vh
+        # We choose the singular values to be "around one"
+        # This is to make the matrix well conditioned
+        # s = [2, 3, ..., k+1]
```

Comment:
A mental note: we can start with any value `k > 2` to produce even better conditioned inputs.

Reply:
imo, `9/4` is a pretty good one and I'd be surprised if it weren't good enough in any of the context we use it in. Good to keep in mind tho.

Reply:
Might as well add a comment